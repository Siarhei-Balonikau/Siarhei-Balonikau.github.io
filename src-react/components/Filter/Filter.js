import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const Filter = props => {
  const options = props.options.map((option, index) => {
    return index === 0 ? (
      <React.Fragment key={index + 1}><option name="none">None</option><option name={option}>{option}</option></React.Fragment>
    ) : (
      <option key={index + 1} name={option}>{option}</option>
    )
  });
  
  return (
    <select value={props.filter} className="filter" onChange={(e) => props.onChange(e.target.value)}>
      {options}
    </select>
  );
};

Filter.propTypes = {
  options: PropTypes.array.isRequired,
  onChange: PropTypes.func,
  filter: PropTypes.string
};

export default Filter;