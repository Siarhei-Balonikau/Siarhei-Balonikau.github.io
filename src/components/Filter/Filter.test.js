import React, { Component } from 'react';
import { shallow } from 'enzyme';
import Filter from './Filter'; 
import toJson from 'enzyme-to-json'

describe('subscribe component', () => {
  let wrapper, props;

  beforeEach(() => {
    props = {
      options: [
        'test',
        'test1'
      ]
    };
  
    wrapper = shallow(<Filter {...props} />);
  });
  
  it('should render component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});