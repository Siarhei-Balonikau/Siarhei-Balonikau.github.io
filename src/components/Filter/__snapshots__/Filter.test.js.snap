// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscribe component should render component 1`] = `
<select
  className="filter"
  onChange={[Function]}
>
  <React.Fragment
    key="1"
  >
    <option
      name="none"
    >
      None
    </option>
    <option
      name="test"
    >
      test
    </option>
  </React.Fragment>
  <option
    key="2"
    name="test1"
  >
    test1
  </option>
</select>
`;
